be/src/exec/pipeline/exchange/mcast_local_exchange.cpp

```diff
@@ -0,0 +1,206 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "exec/pipeline/exchange/mcast_local_exchange.h"
+
+#include "util/logging.h"
+
+namespace starrocks {
+namespace pipeline {
+
+MultiCastLocalExchanger::MultiCastLocalExchanger(size_t consumer_number)
+        : _mutex(),
+          _consumer_number(consumer_number),
+          _current_acc_chunk_size(0),
+          _progress(consumer_number),
+          _opened_source_opcount(consumer_number) {
+    Cell* dummy = new Cell();
+    _head = dummy;
+    _tail = dummy;
+    for (size_t i = 0; i < consumer_number; i++) {
+        _progress[i] = _tail;
+        _opened_source_opcount[i] = 0;
+    }
+}
+
+MultiCastLocalExchanger::~MultiCastLocalExchanger() {
+    while (_head != nullptr) {
+        auto* t = _head->next;
+        delete _head;
+        _head = t;
+    }
+}
+
+bool MultiCastLocalExchanger::can_push_chunk() const {
+    std::unique_lock l(_mutex);
+    if ((_current_acc_chunk_size - _fast_acc_chunk_size) > config::vector_chunk_size) {
+        return false;
+    }
+    return true;
```

Comment:
Can't return false in there, it's will block execute flow, and we should find a better way to interrupt the whole execute flow, the way maybe is checkout memory limit.

Reply:
It will not block. The driver just asks if exchange has chunk. If has chunk, then driver will pull chunk from it.

Reply:
Thanks, I see, plz add some comment